x-pack/plugins/spaces/public/management/edit_space/delete_spaces_button.tsx

```diff
@@ -99,23 +99,24 @@ export class DeleteSpacesButton extends Component<Props, State> {
   public deleteSpaces = async () => {
     const { spacesManager, space } = this.props;
 
+    this.setState({
+      showConfirmDeleteModal: false,
+    });
+
     try {
       await spacesManager.deleteSpace(space);
     } catch (error) {
-      const { message: errorMessage = '' } = error.data || {};
+      const { message: errorMessage = '' } = error.data || error.body || {};
 
       this.props.notifications.toasts.addDanger(
         i18n.translate('xpack.spaces.management.deleteSpacesButton.deleteSpaceErrorTitle', {
           defaultMessage: 'Error deleting space: {errorMessage}',
           values: { errorMessage },
         })
       );
+      return;
```

Comment:
Added this return statement; otherwise, in an error scenario, the "danger" toast is shown and immediately followed by a "success" toast.